### PR TITLE
Rename tags to teams

### PR DESCRIPTION
**Pull request by B1LLP4RK**
Renamed tags to teams in all java files in source and test and UserGuide

Renamed the class Tag into Team. Renamed all variables and methods in all java codes relevant to the tagging feature in AB3 to a name relevant to 'team'. For example, the constant 'INVALID_TAG' in the testing module was renamed to 'INVALID_TEAM'. Relevant sections in UserGuide were modified as well.

### CODE REVIEW

File: src/main/java/seedu/address/model/team/Team.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    public Team(String teamName) {
+        requireNonNull(teamName);
+        checkArgument(isValidteamName(teamName), MESSAGE_CONSTRAINTS);
+        this.teamName = teamName;
+    }
+
+    /**
+     * Returns true if a given string is a valid team name.
+     */
+    public static boolean isValidteamName(String test) {
```
Comment (Jiaaa-yang) [inline src/main/java/seedu/address/model/team/Team.java:31]: I think the naming convention in terms of the camelCase here is slightly off? Maybe `isValidTeamName` or even `isValidName` would be good?

File: src/main/java/seedu/address/storage/JsonAdaptedPerson.java [hunk 2]
```diff
@@ -38,14 +38,14 @@
     @JsonCreator
     public JsonAdaptedPerson(@JsonProperty("name") String name, @JsonProperty("phone") String phone,
             @JsonProperty("email") String email, @JsonProperty("username") String username,
-            @JsonProperty("tagged") List<JsonAdaptedTag> tagged,
+            @JsonProperty("teamed") List<JsonAdaptedTeam> teamSet,
```
Comment (Jiaaa-yang) [inline src/main/java/seedu/address/storage/JsonAdaptedPerson.java:41]: Would the property name 'teams' be better here?

File: src/test/data/JsonSerializableAddressBookTest/duplicatePersonAddressBook.json [hunk 3]
```diff
@@ -4,7 +4,7 @@
     "phone": "94351253",
     "email": "alice@example.com",
     "username": "alice-123",
-    "tagged": [ "friends" ],
+    "teamed": [ "friends" ],
```
Comment (Jiaaa-yang) [inline src/test/data/JsonSerializableAddressBookTest/duplicatePersonAddressBook.json:7]: Same with my previous comment. Maybe `teams` would fit better than `teamed`?

**Review by Jiaaa-yang — APPROVED** [review]
Hey this PR looks good to merge! Have some comments that would not affect implementations which you may want to take a look at! I think its good that you kept the naming conventions of things you refactored from the original tags!

File: src/main/java/seedu/address/model/team/Team.java (hunk 1 above)
Comment (B1LLP4RK) [reply src/main/java/seedu/address/model/team/Team.java:31]: yes I agree 👍

**Pull request merged**